Replace more container ranges with visitor

verb opened the pull request:
These are all no-ops for ephemeral containers since they affect fields not used by ephemeral containers. This is purely a clean up.

**What type of PR is this?**
/kind cleanup
/priority important-longterm
/sig node

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

mattjmcnaughton reviewed (commented):
Similar to my comment here: https://github.com/kubernetes/kubernetes/pull/81838#pullrequestreview-279067278

I'm curious to know more about the motivation behind introducing `VisitContainers`.

verb committed:
Replace more container ranges with visitor

These are all no-ops for ephemeral containers since they affect fields
not used by ephemeral containers. This is purely a clean up.

verb commented:
> Similar to my comment here: [#81838 (review)](https://github.com/kubernetes/kubernetes/pull/81838#pullrequestreview-279067278)
> 
> I'm curious to know more about the motivation behind introducing `VisitContainers`.

In this particular file, it's for consistency. `VisitContainers` is defined here, and the other methods in it use `VisitContainers` for iteration. We can also simplify a bit by replacing specific methods with the generic `VisitContainers`.
